### ISSUE TITLE
Two templates get merged together in listing

### ISSUE THREAD

**Issue by petrroll** [dotnet/templating#649]
I have a problem where two of my templates get merged into one in the `dotnet new -a` listing. A user can still invoke each of them just fine if he knows their (either full or short) names and it even shows both of them in the `Examples` section after calling `dotnet new <their-common-prefix>`. In the listings (`new -a`, `new -l`, ...), however, it shows just one of them.

The templates in question: [web ](https://github.com/peachpiecompiler/peachpie-templates/tree/master/src/Content/web-app) and [console](https://github.com/peachpiecompiler/peachpie-templates/tree/master/src/Content/console-app).

After install listing:
```
PS C:\Users\Petr\Source\Repos\peachpie-templates\build> dotnet new --install .\Peachpie.Templates.0.1.0.nupkg
Getting ready...
Templates                         Short Name            Language      Tags
...
Peachpie console application      peachpie-console      PHP, PHP      Console
...
```

Common prefix invocation:
```
PS C:\Users\Petr\Downloads\tmp> dotnet new peachpie
Unable to determine the desired template from the input template name: [peachpie]
The following templates partially match the input. Please be more specific with the template name and/or language.
        Peachpie console application
        Peachpie web application
...
Templates                         Short Name            Language      Tags
-----------------------------------------------------------------------------
Peachpie console application      peachpie-console      PHP, PHP      Console

Examples:
    dotnet new peachpie-console
    dotnet new peachpie-web
    dotnet new --help
```

Listing:
```
PS C:\Users\Petr\Downloads\tmp> dotnet new -l
...
Templates                         Short Name            Language      Tags
------------------------------------------------------------------------------------
Console Application               console               [C#], F#      Common/Console
Class library                     classlib              [C#], F#      Common/Library
Peachpie console application      peachpie-console      PHP, PHP      Console
Unit Test Project                 mstest                [C#], F#      Test/MSTest
xUnit Test Project                xunit                 [C#], F#      Test/xUnit
ASP.NET Core Empty                web                   [C#]          Web/Empty
ASP.NET Core Web App              mvc                   [C#], F#      Web/MVC
ASP.NET Core Web API              webapi                [C#]          Web/WebAPI
Solution File                     sln                                 Solution
```

**Comment by mlorbetske** [comment]
If the two templates have the same value for groupIdentity, this will happen and is by design. groupIdentity indicates that the templates should display as a single template (a set of equivalent templates for multiple languages is the canonical example of having two with the same groupIdentity). In the 1.0.x CLI there's a bug where all templates that do not specify a groupIdentity get unified, this has been fixed in the 2.0 release.

**Comment by jannikbuschke** [comment]
This is really a weird api. I lived several weeks with broken templates as I am not aware of this. The templates got merged and somehow the resulting thing does not output any files https://github.com/dotnet/templating/issues/1872.
No warnings/errors.
